product-service/src/functions/createProduct/handler.ts

```diff
@@ -0,0 +1,47 @@
+import createError from 'http-errors';
+import { formatJSONResponse, ValidatedEventAPIGatewayProxyEvent } from 'src/libs/apiGateway';
+import { middyfy } from 'src/libs/lambda';
+
+import schema from './schema';
+
+type GetProductByIdParamsType = {
+  productId: string,
+}
+
+export function createProduct({ productId }: GetProductByIdParamsType) {
+  if (!productId) {
+    throw createError(404);
+  }
+
+  return formatJSONResponse(productId);
+}
+
+const handler: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async (event, context) => {
+  const { body: { title, description, price, count } } = event;
+  const { dbClient } = context.clientContext.Custom;
+
+  await dbClient.query('BEGIN;');
+  try {
+    const { rows: [{ id: productId }] } = await dbClient.query(
+      'INSERT INTO products (title, description, price) VALUES ($1, $2, $3) RETURNING id;',
+      [title, description ?? 'NULL', price]
```

Comment:
I believe that `description ?? null` could be enough here to prevent setting empty string. Database by default should set it as 'NULL` in case of absence the value.

Reply:
Neither my code nor yours (`description ?? null`) will prevent it from setting an empty string)
It could be done by `description || null`. But I don't think it matters. I'll better remove the default at all

Reply:
You are right. Operator `??` check only on absence of value (`null` or `undefined`). Holidays are not good for me 😄 .